cargo-nextest/src/dispatch.rs

```diff
@@ -412,6 +454,44 @@ impl TestBuildFilter {
                 .map_err(|err| ExpectedError::CreateTestListError { err })?,
         )
     }
+
+    fn merge_test_binary_args(&mut self, args: &mut Vec<String>) -> Result<()> {
+        let mut ignore_filters = Vec::new();
+        args.retain(|s| {
+            if s == "--include-ignored" {
+                ignore_filters.push((s.clone(), RunIgnored::All));
+                false
+            } else if s == "--ignored" {
+                ignore_filters.push((s.clone(), RunIgnored::IgnoredOnly));
+                false
+            } else if !s.starts_with('-') {
+                self.filter.push(s.to_owned());
+                false
+            } else if s.chars().all(|c| c == '-') {
+                false
```

Comment:
This is still not quite there:
1. This needs to just be `--` (a single `-` doesn't match, nor does `---`).
2. Once this is hit, it should go into a mode where the `s.starts_with('-')` check is disabled and everything is pushed into the filter.